stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/composer/MessageComposerView.kt

```diff
@@ -0,0 +1,195 @@
+            override fun afterTextChanged(s: Editable) {
+                onInputTextChanged(s.toString())
+            }
+        })
+        binding.clearCommandButton.isVisible = false
+        binding.selectedAttachmentsContainer.isVisible = false
+        binding.commandBadge.isVisible = false
+        binding.messageReplyView.isVisible = false
+        CoroutineScope(Dispatchers.Main).launch {
+            messageInputState.collect {
+                binding.messageEditText.apply {
+                    val currentValue = text.toString()
+                    val newValue = it.inputValue
+                    if (newValue != currentValue) {
+                        setText(it.inputValue)
+                    }
+                }
+                binding.clearCommandButton.isVisible = it.inputValue.isNotEmpty()
+            }
+        }
+        binding.root
+    }
+
+    /**
+     * @property centerContent A function returning content visible in the middle.
+     */
+    public var centerContent: MessageComposerView.() -> View = {
+        defaultCenterContent
+    }
+        set(value) {
+            field = value
+            binding.centerContent.apply {
+                removeAllViews()
+                addView(value())
+            }
+        }
+
+    private val defaultTrailingContent: View by lazy {
+        val binding = StreamUiMessageComposerDefaultTrailingContentBinding.inflate(
+            streamThemeInflater,
+            binding.trailingContent,
+            false
+        ).apply {
+            CoroutineScope(Dispatchers.Main).launch {
+                messageInputState.collect {
+                    val sendButtonEnabled = it.inputValue.isNotEmpty()
+                    sendMessageButtonDisabled.isVisible = !sendButtonEnabled
+                    sendMessageButtonEnabled.isVisible = sendButtonEnabled
+                }
+            }
+
+            sendMessageButtonEnabled.setOnClickListener {
+                onSendMessageAction()
+            }
+        }
+        binding.root
+    }
+
+    /**
+     * @property trailingContent A function returning content visible on the right.
+     */
+    public var trailingContent: MessageComposerView.() -> View = { defaultTrailingContent }
+        set(value) {
+            field = value
+            binding.trailingContent.apply {
+                removeAllViews()
+                addView(value())
+            }
+        }
+
+    public constructor(context: Context) : this(context, null)
+
+    public constructor(context: Context, attrs: AttributeSet?) : this(context, attrs, 0)
+
+    public constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context,
+        attrs,
+        defStyleAttr) {
+        init()
+    }
+
+    private fun init() {
+        binding = StreamUiMessageComposerBinding.inflate(streamThemeInflater, this)
+        binding.leadingContent.apply {
+            removeAllViews()
```

Comment:
Should we maybe check if there are existing views and everything is set up, before calling removing all views and re-adding them?

In case of configuration changes, this will just reset all the state, right? That's also why having the MessageInputState in the ViewModel is kind of better, because it's persisted across orientation changes.